malariagen_data/ag3.py

```diff
@@ -484,10 +528,17 @@ def open_snp_genotypes(self, sample_set):
             self._cache_snp_genotypes[sample_set] = root
             return root
 
+    def _snp_genotypes(self, *, contig, sample_set, field, inline_array, chunks):
```

Comment:
What does the `*` do here?

Reply:
It disallows any positional arguments, i.e., forces the method to be called with keyword only arguments. This is defensive, a common source of bugs is where you rely on positional arguments but then later add more positional args or change the argument order, and forget to update calling code somewhere. Forcing all calls to use keyword only arguments means if you add more arguments then all previous code will still work. See also https://www.python.org/dev/peps/pep-3102/.